pkg/sysregistriesv2/shortnames.go

```diff
@@ -0,0 +1,233 @@
+package sysregistriesv2
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/homedir"
+	"github.com/containers/storage/pkg/lockfile"
+	"github.com/pkg/errors"
+)
+
+// userShortNamesFile is the user-specific registries.conf file to store aliases.
+var userShortNamesFile = filepath.FromSlash(".config/containers/short-name-alias.conf")
+
+// shortNameAliasConf is a subset of the `V2RegistriesConf` format.  It's used in the
+// software-maintained `userShortNamesFile`.
+type shortNameAliasConf struct {
+	// A map for aliasing short names to their fully-qualified image
+	// reference counter parts.
+	Aliases map[string]string `toml:"aliases"`
+}
+
+// ShortNameMode defines how short-name resolution should be handled by
+// _consumers_ of this package.
+type ShortNameMode int
+
+const (
+	// Do not prompt.
+	ShortNameModeDisabled = iota // we default to it for backwards compat!
```

Comment:
`ShortNameMode{Invalid,Uninitialized} ShortNameMode = iota`, to
- Make the constants well-typed
- Detect uses of uninitialized values